core/src/main/java/org/infinispan/remoting/transport/jgroups/FilterMapResponseCollector.java

```diff
@@ -0,0 +1,65 @@
+package org.infinispan.remoting.transport.jgroups;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.infinispan.commands.ReplicableCommand;
+import org.infinispan.commons.util.CollectionFactory;
+import org.infinispan.remoting.inboundhandler.DeliverOrder;
+import org.infinispan.remoting.responses.CacheNotFoundResponse;
+import org.infinispan.remoting.responses.Response;
+import org.infinispan.remoting.responses.ValidResponse;
+import org.infinispan.remoting.rpc.ResponseFilter;
+import org.infinispan.remoting.rpc.ResponseMode;
+import org.infinispan.remoting.transport.Address;
+import org.infinispan.remoting.transport.ValidResponseCollector;
+
+/**
+ * Response collector supporting {@link JGroupsTransport#invokeRemotelyAsync(Collection, ReplicableCommand, ResponseMode, long, ResponseFilter, DeliverOrder, boolean)}.
+ *
+ * <p>This class is not thread-safe by itself. It expects an {@link org.infinispan.remoting.transport.AbstractRequest}
+ * to handle synchronization.</p>
+ */
+public class FilterMapResponseCollector extends ValidResponseCollector<Map<Address, Response>> {
```

Comment:
I do want to remove the `ResponseFilter`, but I'm not sure `ResponseCollector` is stable enough yet. So I'm thinking of marking the `ResponseCollector`-based methods as experimental in 9.1 and only deprecating the `ResponseFilter` ones in 9.2/10.0.